clusterloader2/pkg/framework/client/objects.go

```diff
@@ -162,10 +168,13 @@ func PatchObject(dynamicClient dynamic.Interface, namespace string, name string,
 		if err != nil {
 			return fmt.Errorf("creating patch diff error: %v", err)
 		}
-		_, err = dynamicClient.Resource(gvr).Namespace(namespace).Patch(obj.GetName(), types.StrategicMergePatchType, patch)
+		newObj, err = dynamicClient.Resource(gvr).Namespace(namespace).Patch(obj.GetName(), types.StrategicMergePatchType, patch)
 		return err
 	}
-	return RetryWithExponentialBackOff(retryFunction(updateFunc, nil))
+	if err := RetryWithExponentialBackOff(retryFunction(updateFunc, nil)); err != nil {
+		return nil, err
+	}
+	return newObj, nil
```

Comment:
This approach has a visible problem, that we don't have a way to detect RV of Delete operation (which is pretty crucial) - we want to track that too.

And I don't see a good solution for that now.
And it seems that we may have over-engineered that.

After thinking about that more, I think we may significantly simplify that by:
- not tracking RVs at state level at all
- when you call "gather" in the measurement you:
 (a) first call "LIST" on that kind (say RCs) and store the RV from the ListMeta
 (b) wait in the watch until it's synced with that RV
 (c) do the real waiting for all of the started go-routines

Sorry for introducing unnecessary complications - the above seems to be much easier and doesn't have the problem with deletions (and doesn't require changes all over the code).